src/controllers/restaurantsController.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from "express";
+
+import pool from "../db.js";
+import restaurantsNearYouSchema from "../schemas/restaurantsNearYouReq.js";
+import { sendResponse } from "../helpers/responses.js";
+import logError from "../helpers/logger.js";
+import Status from "../helpers/types/status.js";
+import { Operation } from "../helpers/types/operation.js";
+
+const defaultRange = Number(process.env.DEFAULT_RANGE);
+
+export const getRestaurantsNearYou = async (req: Request, res: Response) => {
+  const validatedRequest = restaurantsNearYouSchema.safeParse(req.query);
+
+  if (!validatedRequest.success) {
+    return sendResponse(
+      res,
+      Status.Fail,
+      "Invalid request",
+      Operation.BadRequest
+    );
+  }
+
+  // Convert all the query values into numbers
+  const range = validatedRequest.data.range
+    ? Number(validatedRequest.data.range)
+    : defaultRange;
+  const userLatitude = Number(validatedRequest.data.user_lat);
```

Comment:
When you Number() an empty string it coerces it to 0.

After this you check if isNaN which 0 is a number so it passes this check too.

You can fix this with my other comment in your schema, by making sure empty strings are caught in the zod parsing